Localized headers

danielebogo opened the pull request:
Fixes PCIOS-170

[Android ref.](https://github.com/Automattic/pocket-casts-android/pull/4580)

This PR adds the `App-Language` and `User-Region` to our main requests.

Requests not modified:
```
MediaExporterResourceLoaderDelegate (line 186)
PodcastChapterParser (line 86)
TokenHelper
```

@bjtitus @SergioEstevao do we need these?

## To test

1. Review each commit to double check the modified requests are correct.
2. Start the app.
3. If you can log our network request with some tool use it, otherwise set the breakpoint to check the header status
4. The headers should match your configuration.
5. Go to Discover and scroll to the bottom.
6. Change your region.
7. Value should be reflected in the `X-User-Region` header.
8. Change device's or app's language from the system settings.
9. Value should be reflected in the `X-App-Language` header.

## Checklist

- [ ] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

danielebogo committed:
Add FF

danielebogo committed:
Add Localizations Helper

danielebogo committed:
Add Helper

danielebogo committed:
Add header to ApiBaseTask

danielebogo committed:
Add header to ServerHelper

danielebogo committed:
Add header to MetadataTask

danielebogo committed:
Add header to CacheServerHandler

danielebogo committed:
Add header to ShowInfoDataRetriever

danielebogo committed:
Add header to UploadManager

danielebogo committed:
Add header to ServerPodcastManager

danielebogo committed:
Update custom headers value

bjtitus reviewed (approved):
:shipit: 

`MediaExporterResourceLoaderDelegate` and `PodcastChapterParser` are both external URLs so it's probably not an issue.

`TokenHelper` could be a spot to add to all requests at once but it's probably fine to add it to more specific requests if that makes more sense.

MiSikora commented:
@danielebogo I'm going to change Android implementation to use headers only with our services. See: https://github.com/Automattic/pocket-casts-android/pull/4580#discussion_r2412717817

danielebogo commented:
Thanks @MiSikora !

danielebogo committed:
Add allowed hosts and tests

danielebogo committed:
Use FF in app

danielebogo committed:
Add teardown to tests

danielebogo committed:
Add localization to token related requests

danielebogo commented:
@bjtitus I followed the Android approach mentioned above to list the allowed hosts.

I added few unit tests and added the localization to few requests related to the `TokenHelper` [here](https://github.com/Automattic/pocket-casts-ios/pull/3553/commits/ac1822acea9a0f340323d58f16c8796aa4bbe85a)

danielebogo committed:
Add localization to missing search tasks

bjtitus reviewed (commented):
LGTM :shipit:

SergioEstevao reviewed (approved):
All working correctly.
I used a proxy (Proxyman) to look at the request being sent and I could see the localization headers on the requested services. Also smoke tested the app and all is looking correct.

danielebogo committed:
Merge branch 'trunk' into danieleb/PCIOS-170-localized-headers

# Conflicts:
#	Modules/Utils/Sources/PocketCastsUtils/Feature Flags/FeatureFlag.swift
